src/datafusion/config.rs

```diff
@@ -19,14 +19,40 @@ use std::sync::Arc;
 
 use object_store::ObjectStore;
 
+const DEFAULT_BATCH_SIZE: usize = 8096;
+
+/// Configuration for Zarr DataFusion processing.
 #[derive(Clone)]
 pub struct ZarrConfig {
-    // The object store to use.
+    /// The object store to use.
     pub object_store: Arc<dyn ObjectStore>,
+
+    /// The projection for the scan.
+    pub projection: Option<Vec<usize>>,
+
+    /// The batch size for the scan.
+    pub batch_size: usize,
```

Comment:
So is this used in the code you wrote yet? I don't think I see it, I might have missed it, but in general, I think for certain things we'll have to do things differently for zarr (compared to, say, a parquet file opener). In particular, I'm not too sure the concept of a batch size works here, I was thinking the batches would come directly from the zarr chunks.

I'm curious though, how would you see batch sizes working here? Considering that the data is split between potentially a lot of files (100s, potentially 1000s), and more importantly that the data will likely be compressed? For the parquet implementation, which I think doesn't handle compressed parquet files (I might be wrong), and where a lot of data could be stored in a single file, reading batches of rows makes sense, but I can't quite wrap my head around how it would work here, for zarr files.